### PR TITLE
audit tests

### PR DESCRIPTION
**Pull request by blasrodri**
This is awesome @dhruvja ! 

My first audit is on the tests. I've left a bunch of comments with `TODO`. Overall they look good.
- Add a few helper functions to make the code more readable and clean some boilerplate of loading pdas from the main business logic
- implement missing tests (failed cases  + validation of amount of rewards)

When this is done, I"ll move to the Rust code. But I can quickly say that it needs to have waaaay more docs on each instruction. As a reference check

### CODE REVIEW

File: tests/candidate_staking.ts [hunk 1]
```diff
@@ -575,6 +565,12 @@ describe("candidate_staking", () => {
     assert.equal(_casTokenWallet.amount, initialMintAmount - stakeAmount);
   });
 
+  /*
```
Comment (blasrodri) [inline tests/candidate_staking.ts:568]: add these tests

Comment (dhruvja) [reply tests/candidate_staking.ts:568]: ok will add them

**Review by dhruvja — COMMENTED** [review]
Can u pull the changes from the latest update from master and push?

**Comment by blasrodri** [comment]
merge this

**Pull request merged**